presto-jdbc/src/test/java/io/prestosql/jdbc/BaseTestJdbcResultSet.java

```diff
@@ -90,6 +118,40 @@ public void testDecimal()
         try (ConnectedStatement connectedStatement = newStatement()) {
             checkRepresentation(connectedStatement.getStatement(), "0.1", Types.DECIMAL, new BigDecimal("0.1"));
             checkRepresentation(connectedStatement.getStatement(), "DECIMAL '0.1'", Types.DECIMAL, new BigDecimal("0.1"));
+            checkRepresentation(connectedStatement.getStatement(), "0.12", Types.DECIMAL,
+                    (rs, column) -> assertEquals(rs.getDouble(column), 0.12D));
+            checkRepresentation(connectedStatement.getStatement(), "DECIMAL '0.1'", Types.DECIMAL,
+                    (rs, column) -> assertEquals(rs.getDouble(column), 0.1D));
+            checkRepresentation(connectedStatement.getStatement(), "DECIMAL '0.1'", Types.DECIMAL,
+                    (rs, column) -> assertEquals(rs.getLong(column), 0));
```

Comment:
Combine them into a single case ("0.12")

Also, no need to use `D` for double literals in Java (they are double by default).

Would be good to add assertion for rs.getFloat(..) too